### PR TITLE
New distortion

### PR DESCRIPTION
**Pull request by joelvbernier**
This PR has the new distortion package implemented.  Tested instrument initialization, and applicaiton via the hexrdgui in polarview.

### CODE REVIEW

File: hexrd/instrument.py [hunk 1]
```diff
@@ -1832,10 +1835,7 @@ def distortion(self):
 
     @distortion.setter
     def distortion(self, x):
-        """
-        Probably should make distortion a class...
-        ***FIX THIS***
-        """
+        # FIXME: ne to reconcile check with new class type!
         assert len(x) == 2 and hasattr(x[0], '__call__'), \
```
Comment (joelvbernier) [inline hexrd/instrument.py:1839]: incorrect assertion for now, need to fix this setter.

File: hexrd/xrdutil.py [hunk 2]
```diff
@@ -774,8 +771,8 @@ def angularPixelSize(
         * assumes xy_det in UNWARPED configuration
         """
         xy_det = np.atleast_2d(xy_det)
-        if distortion is not None and len(distortion) == 2:
-            xy_det = distortion[0](xy_det, distortion[1])
+        if distortion is not None:  # !!! check this logic
```
Comment (joelvbernier) [inline hexrd/xrdutil.py:774]: Seems ok.

**Comment by joelvbernier** [comment]
@psavery -- didn't mean to burden you with this.  I am testing and it seems ok for now (espeially for such a big change!).

File: hexrd/transforms/xf.py [hunk 3]
```diff
@@ -11,9 +11,9 @@
 #
```
Comment (joelvbernier) [inline hexrd/transforms/xf.py:11]: wtf is wrong with me; we don't even use this module!

**Comment by psavery** [comment]
I'm gonna take a look today. Sorry for the delay...

**Comment by joelvbernier** [comment]
This looks to be ok, and the places it touched are isolated to the overlays and the calibration utilities.  The latter are going to be moved into hexrd, and the former actually need to be modified to even use distortion anyhow (only affects the raw view when distortion is not `None`).  We can do that in a future PR.

**Pull request merged**